include/wil/chrono.h

```diff
@@ -0,0 +1,374 @@
+            using duration = std::chrono::duration<rep, period>;
+            using time_point = std::chrono::time_point<base_clock_t, duration>;
+            static constexpr bool is_steady = true;
+
+            WI_NODISCARD static time_point now() WI_NOEXCEPT
+            {
+                ULONGLONG t;
+                query_interrupt_time_fn(&t);
+                return time_point{duration{ static_cast<rep>(t) }};
+            }
+        };
+    }
+
+#if _WIN32_WINNT >= 0x0601
+    struct unbiased_interrupt_time_clock : details::interrupt_time_clock_impl<decltype(&::QueryUnbiasedInterruptTime), ::QueryUnbiasedInterruptTime, unbiased_interrupt_time_clock> {};
+#if defined(NTDDI_WIN10)
+    struct interrupt_time_clock : details::interrupt_time_clock_impl<decltype(&::QueryInterruptTime), ::QueryInterruptTime, interrupt_time_clock> {};
+    struct precise_interrupt_time_clock : details::interrupt_time_clock_impl<decltype(&::QueryInterruptTimePrecise), ::QueryInterruptTimePrecise, interrupt_time_clock> {};
+    struct precise_unbiased_interrupt_time_clock : details::interrupt_time_clock_impl<decltype(&::QueryUnbiasedInterruptTimePrecise), ::QueryUnbiasedInterruptTimePrecise, unbiased_interrupt_time_clock> {};
+#endif // defined(NTDDI_WIN10)
+#endif // _WIN32_WINNT >= 0x0601
+#pragma endregion
+
+#pragma region std::chrono wrappers for thread/process times
+    using cpu_time_duration = std::chrono::duration<LONGLONG, std::ratio_multiply<std::hecto, std::nano>>;
+
+    enum class cpu_time
+    {
+        total,
+        kernel,
+        user,
+    };
+
+    struct execution_times
+    {
+        system_time_clock::time_point creation_time;
+        system_time_clock::time_point exit_time;
+        cpu_time_duration kernel_time;
+        cpu_time_duration user_time;
+    };
+
+    namespace details
+    {
+        inline __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 execution_times execution_times_from_filetimes(FILETIME creation_time, FILETIME exit_time, FILETIME kernel_time, FILETIME user_time) WI_NOEXCEPT
+        {
+            return {
+                system_time_clock::from_filetime(creation_time),
+                system_time_clock::from_filetime(exit_time),
+                cpu_time_duration{ static_cast<cpu_time_duration::rep>(details::filetime_to_int(kernel_time)) },
+                cpu_time_duration{ static_cast<cpu_time_duration::rep>(details::filetime_to_int(user_time)) }
+            };
+        }
+
+        inline __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 cpu_time_duration get_cpu_time(const execution_times& times, cpu_time kind) WI_NOEXCEPT
+        {
+            switch (kind)
+            {
+                case cpu_time::total: return times.kernel_time + times.user_time;
+                case cpu_time::kernel: return times.kernel_time;
+                case cpu_time::user: return times.user_time;
+                DEFAULT_UNREACHABLE;
+            }
+        }
+    }
+
+    using thread_times = execution_times;
+
+    template<class error_policy>

```

Comment:
It feels like these should get broken up in to the three `foo`, `foo_nothrow`, and `foo_failfast` variants since there's no way for the 'nothrow' instantiation to report failure. Seems like it'll just return values from (possibly) uninitialized memory.